fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/CascadesCostModel.java

```diff
@@ -119,8 +120,22 @@ public int compare(@Nonnull RelationalExpression a, @Nonnull RelationalExpressio
         if (distinctFilterPositionCompare != 0) {
             return distinctFilterPositionCompare;
         }
-        return Integer.compare(UnmatchedFieldsProperty.evaluate(planContext, a),
-                UnmatchedFieldsProperty.evaluate(planContext, b));
+
+        int ufpA = UnmatchedFieldsProperty.evaluate(planContext, a);
+        int ufpB = UnmatchedFieldsProperty.evaluate(planContext, b);
+        if (ufpA != ufpB) {
+            return Integer.compare(ufpA, ufpB);
+        }
+
+        // If plans are indistinguishable from a cost perspective, select one by planHash. This would make the cost model stable
+        // (select the same plan on subsequent plannings).
+        if ((a instanceof PlanHashable) && (b instanceof PlanHashable)) {
```

Comment:
is there any chance this is not a `PlanHashable`

Reply:
In my tests I have not seen it be called with non-PlanHashable items but for defensive coding decided it may be better to check.